### PR TITLE
Set worker_processes to auto in nginx.conf

### PR DESCRIPTION
**Pull request by mecampbellsoup**
We can let NGINX determine the optimal number of worker process to run instead of limiting it to 1 worker per master.

image

https://nginx.org/en/docs/ngx_core_module.html#worker_processes
https://www.nginx.com/blog/inside-nginx-how-we-designed-for-performance-scale/

### CODE REVIEW

**Comment by salanki** [comment]
No. We run multiple nginexes per node, you don't want each nginx to spin up a gazillion processes. It should stay at 1.

**Comment by mecampbellsoup** [comment]
My argument is we should run one NGINX process w/ `n` workers instead. We don't need multiple NGINX master processes, that is extremely redundant and super inefficient as they all cache the same stuff in memory for their respective workers.

**Review by kriscode1 — CHANGES_REQUESTED** [review]
Nginx uses [sysconf(_SC_NPROCESSORS_ONLN)](https://www.man7.org/linux/man-pages/man3/sysconf.3.html) to detect the number of cores available. A quick test shows that this core count doesn't change even when inside a container with limited CPUs (that's a container-engine concept). So nginx will see 8 cores, for example, but may only have 100milliCPU of time. Reading the nginx docs above, it's best to set the requested resources of the pod to an integer number of CPUs and then set `worker_processes` to the same number.

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+envsubst '${WORKER_PROCS}' < /etc/nginx/templates/nginx.conf.template > /etc/nginx/nginx.conf
```
Comment (kriscode1) [inline entrypoint.sh:3]: Nifty!

**Review by kriscode1 — APPROVED** [review]

**Pull request merged**